src/Serializer/JsonApiSerializer.php

```diff
@@ -72,6 +72,11 @@ public function item($resourceKey, array $data)
 
         unset($resource['data']['attributes']['id']);
 
+        if(isset($resource['data']['attributes']['links'])) {
+            $resource['data']['links'] = $data['links'];
+            unset($resource['data']['attributes']['links']);
+        }
+
         if ($this->shouldIncludeLinks()) {
             $resource['data']['links'] = [
                 'self' => "{$this->baseUrl}/$resourceKey/$id",
```

Comment:
@PinkTiu Your issue is correct, your solution is not. :) I think I fixed it now.